src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -29,27 +28,25 @@ public static class TarUtils
             string inputPath,
             CancellationToken cancellationToken)
         {
-            var archiveFile = Path.Combine(Path.GetTempPath(), archiveFileName);
-            if (File.Exists(archiveFile))
+            if(File.Exists(inputPath))
             {
-                File.Delete(archiveFile);
+                throw new InvalidOperationException($"Please specify path to a directory, File path is not allowed. {inputPath} is a file.");
```

Comment:
ArgumentException

Reply:
Or DirectoryNotFound

---
In reply to: [320528839](https://github.com/microsoft/azure-pipelines-agent/pull/2411#discussion_r320528839) [](ancestors = 320528839)